roles/pulp_api/tasks/main.yml

```diff
@@ -62,17 +62,30 @@
   when:
     - (ansible_version.major > 2) or (ansible_version.major == 2 and ansible_version.minor >= 9)
 
+- name: Get database config
+  command: "cat {{ __pulp_db_fields_key_path }}"
+  delegate_to: "{{ __pulp_real_sole_database_config_host }}"
+  register: pulp_db_fields_key
+  changed_when: false
+  become: true
+
+- name: Check for existing Pulp Database Encryption Key
+  stat:
+    path: '{{ __pulp_db_fields_key_path }}'
+    get_attributes: false
+    get_checksum: false
+    get_mime: false
+  register: __pulp_db_fields_key_file
+  become: true
+
 - name: Import Pulp Database Encryption Key
-  copy:
-    src: "{{ pulp_db_fields_key }}"
-    dest: "{{ __pulp_db_fields_key_path }}"
-    owner: 'root'
-    group: "{{ pulp_group }}"
-    mode: 0640
-    remote_src: "{{ pulp_db_fields_key_remote }}"
+  shell: >-
```

Comment:
How about using the `copy` module with the `content` field instead?
https://docs.ansible.com/ansible/latest/collections/ansible/builtin/copy_module.html

This would avoid the need for the `stat`

Just be mindful of the newlines. So Make sure the keyfiles are identical (hashsum of file) across machines.